File: src/main/kotlin/com/huawei/cangjie/diagnostics/SimpleDiagnostic.java
package com.huawei.cangjie.diagnostics;


import com.intellij.psi.PsiElement;
import org.jetbrains.annotations.NotNull;

public class SimpleDiagnostic<E extends PsiElement> extends AbstractDiagnostic<E> {
    public SimpleDiagnostic(
            @NotNull E psiElement,
            @NotNull DiagnosticFactory0<E> factory,
            @NotNull Severity severity
    ) {
        super(psiElement, factory, severity);
    }

    @NotNull
    @Override
    @SuppressWarnings("unchecked")
    public DiagnosticFactory0<E> getFactory() {
        return (DiagnosticFactory0<E>) super.getFactory();
    }
}

File: src/main/kotlin/com/linqingying/cangjie/diagnostics/SimpleDiagnostic.kt
package com.linqingying.cangjie.diagnostics

import com.intellij.psi.PsiElement


class SimpleDiagnostic<E : PsiElement >(
    psiElement: E,
    factory: DiagnosticFactory0<E>,
    severity: Severity
) : AbstractDiagnostic<E>(psiElement, factory, severity) {
    override val factory: DiagnosticFactory0<E>
        get() = super.factory as DiagnosticFactory0<E>
}
